### ISSUE TITLE
Deprecating `strides` in favor of `stride` (or vice versa)

### ISSUE THREAD

**Issue by bicycle1885** [JuliaLang/julia#34922]
I sometimes confuse `stride` with `strides`; these two functions basically do the same thing (i.e., `stride(A, k) = strides(A)[k]`) but have different names. So, I think we can unify them into a single function instead of having both.

I think there are two directions of the unification. One direction is keeping `stride` (i.e., singular) and deprecating `strides`, which aligns well with the `size` function. The other direction is keeping `strides` (i.e., plural) and deprecating `stride`, which aligns well with the `axes` function.
```
julia> A = zeros(2,3,4);

julia> size(A)
(2, 3, 4)

julia> size(A, 1), size(A, 2), size(A, 3)
(2, 3, 4)

julia> axes(A)
(Base.OneTo(2), Base.OneTo(3), Base.OneTo(4))

julia> axes(A, 1), axes(A, 2), axes(A, 3)
(Base.OneTo(2), Base.OneTo(3), Base.OneTo(4))
```

I have no strong opinions on which is better (or you may dislike the idea). So, I'd like to see your ideas and thoughts.

**Comment by StefanKarpinski** [comment]
I guess I'd favor `stride` over `strides`. Asking for all the strides with `stride(A)` feels a bit weird grammatically. Maybe we can just make `strides` and alias for `stride` so that you can write either?

**Comment by bicycle1885** [comment]
Agreed. `stride(A)` to get multiple strides looks grammatically wrong. So I'm slightly in support of keeping `strides` and silently deprecating `stride` so that people can forget it. Aliasing `const stride = strides` sounds a good idea.

**Comment by bramtayl** [comment]
Should `size` be `sizes` then?

**Comment by bramtayl** [comment]
I guess in English sizes is like fishes: you only use the plural for heterogeneous groups. This is why every programming language should be written in Esperanto

**Comment by bicycle1885** [comment]
> This is why every programming language should be written in Esperanto

Or in Japanese, because you don't need to care about singular or plural :stuck_out_tongue_winking_eye:

Anyway, `size(A)` that returns a tuple feels quite natural since it is the size of that array even if it is multidimensional. But stride is not an attribute of an array, it is an attribute for a specific direction of the array.

**Comment by musm** [comment]
I'm definitely in support of deprecated stride in favor of `strides`. Having both is  simply redundant.
